### PR TITLE
feat: add chain_changed event

### PR DESCRIPTION
**Pull request by just-toby**
Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) and start with a valid [type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type).

## Background
https://linear.app/uniswap/issue/WEB-2660/log-chain-changed-events

## Changes
-  add CHAIN_CHANGED event to `InterfaceEventName`

## Testing

👀

### CODE REVIEW

File: src/interface/events.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@
 export enum InterfaceEventName {
   ACCOUNT_DROPDOWN_BUTTON_CLICKED = 'Account Dropdown Button Clicked',
   APPROVE_TOKEN_TXN_SUBMITTED = 'Approve Token Transaction Submitted',
+  CHAIN_CHANGED = 'Chain Changed',
```
Comment (just-toby) [inline src/interface/events.ts:7]: logged when we detect the connected chain ID changes

File: src/interface/events.ts [hunk 2]
```diff
@@ -16,6 +17,7 @@ export enum InterfaceEventName {
   NAVBAR_SEARCH_EXITED = 'Navbar Search Exited',
   NAVBAR_SEARCH_SELECTED = 'Navbar Search Selected',
   RISK_CHECKED = 'Risk Checked',
+  SELECT_CHAIN_INITIATED = 'Select Chain Initiated',
```
Comment (just-toby) [inline src/interface/events.ts:20]: logged when the user selects a chain in the chain selector

**Review by kristiehuang — APPROVED** [review]
loll thank u for ez first PR review!! Lgtm

File: src/interface/events.ts (hunk 2 above)
Comment (kristiehuang) [reply src/interface/events.ts:20]: do we also want event for network switcher dropdown clicked? per Linear ticket

Comment (adjkant) [reply src/interface/events.ts:20]: Can we use an element click event for this one?

**Review by adjkant — APPROVED** [review]

**Pull request merged**